lib/mix/lib/mix/utils.ex

```diff
@@ -326,8 +323,12 @@ defmodule Mix.Utils do
 
   defp do_symlink_or_copy(source, target, link) do
     case :file.make_symlink(link, target) do
-      :ok -> :ok
-      {:error, _} -> {:ok, File.cp_r!(source, target)}
+      :ok ->
+        :ok
+      {:error, _} ->
+        {:ok, File.cp_r!(source, target, fn(orig, dest) ->
+          if (File.stat!(orig).mtime > File.stat!(dest).mtime), do: true, else: false
+        end)}
```

Comment:
`if (...), do: true, else: false`, can we maybe make this `File.stat!(orig).mtime > File.stat!(dest).mtime`?